crates/sui-framework/sources/priority_queue.move

```diff
@@ -128,6 +158,28 @@ module sui::priority_queue {
         check_pop_max(&mut h, 1, 30);
     }
 
+    #[test]
+    fun test_swap_remove_edge_case() {
```

Comment:
I'm hoping it's possible to make this test case more minimal, so that if it fails it will be easier to see what's relevant to that failure.  An easy first step is to get rid of the different values, they can all be zero, but the real benefit comes from needing fewer entries.

I tried to sketch out some small examples to do this, but the heap structure is remarkably resilient (as I'm sure you have already discovered), but I'll keep trying.

This test would also benefit from a comment explaining what it's testing.

Reply:
Here's the smallest example I could come up with:

- `8 7 3 6 2 1 0 5 4`

With successive (incorrectly implemented) pops resulting in:

- `8 : 7 3 6 2 1 0 5 4`
- `7 : 3 6 2 1 0 5 4`
-  `_ : 6 3 2 1 0 5 4` (heapify correcting the top of the heap, but stopping there)
- `6 : 3 2 1 0 5 4`
- `3 : 2 1 0 5 4`

At which point, we've popped 3 and left 4 ad 5 in the heap.  This should hopefully be simpler to follow for someone who hits up against this test failing in future.